code-challenges.test.js

```diff
@@ -39,9 +77,43 @@ const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
 const letterE = "e"
 // Expected output: ["Cherry", "Blueberry", "Peach"]
 
-
+// a describe method that lists the name of the function OR naming of the particular test.
+describe("findThatWord", () => {
+    let arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+    let letterA = "a"
+    let arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+    let letterE = "e"
+    // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+    it("takes in an array of words and a single letter and returns all the words that contain that particular letter.", () => {
+
+        //an expect method, nested within the test block, calling on the findThatWord() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+        expect(findThatWord(arrayOfWords1,letterA)).toEqual(["Apple", "Banana", "Orange"])
+        expect(findThatWord(arrayOfWords2,letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+})
+// -- Got the red --
 
 // b) Create the function that makes the test pass.
+//decalare the function and tales the array as param
+function findThatWord(array, string){
+    //decalare an empty temporary array
+    let tempArr = []
+    //use for loop to search the array param
+    for (let i = 0; i < array.length; i++) {
+      //slipt the array param save in the splitArr
+        let splitArr = array[i].split('')
+        //use for loop to search the splitArr
+        for (let j = 0; j < splitArr.length; j++) {
+          //if the letter's LowerCase equals string param
+          if (splitArr[j].toLowerCase()=== string){
+            //then push the array param's element in to the temporary array
+            tempArr.push(array[i])
+          }
+        }
+    }
+    //return the value, the array has no duplicate values
+    return [...new Set(tempArr)]
+}
```

Comment:
I really appreciate your pseudo code. Nice job. I think you may have done a little more work that is necessary. Some built in methods might help you out.